src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,37 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return new ArrayList<>(new ProducerBulldozer().get());
+        } else if (type == Excavator.class) {
```

Comment:
do we need else block after return statement?

Reply:
Changed too... =)